dns.go

```diff
@@ -242,7 +242,7 @@ const specialHostIP = "169.254.77.65"
 
 // this map contains hardcoded DNS names
 var specialAddresses = map[string]net.IP{
-	specialHostName + ".": net.IP{169, 254, 77, 65},
+	specialHostName + ".": {169, 254, 77, 65},
```

Comment:
I'm going to merge this PR but I have the sense that the readability of this particular line actually deceases with this change, so I'm not 100% sold on using gofumpt across the board for this repo

Reply:
I like gofumpt, but I don't think it's its fault here

I would recommend using https://pkg.go.dev/net#IPv4 here